source/gloperate/include/gloperate/stages/demos/ColorizeStage.h

```diff
@@ -87,13 +86,13 @@ class GLOPERATE_API ColorizeStage : public Stage
 
 protected:
     // Rendering objects
-    gloperate::Camera                          m_camera;
-    gloperate::ScreenAlignedQuad *             m_screenAlignedQuad;
-    globjects::ref_ptr<globjects::VertexArray> m_vao;
-    globjects::ref_ptr<globjects::Buffer>      m_vertexBuffer;
-    globjects::ref_ptr<globjects::Program>     m_program;
-    globjects::ref_ptr<globjects::Shader>      m_vertexShader;
-    globjects::ref_ptr<globjects::Shader>      m_fragmentShader;
+    gloperate::Camera                               m_camera;
+    std::unique_ptr<gloperate::ScreenAlignedQuad>   m_screenAlignedQuad;
+    std::unique_ptr<globjects::VertexArray>         m_vao;
+    std::unique_ptr<globjects::Buffer>              m_vertexBuffer;
+    std::unique_ptr<globjects::Program>             m_program;
+    std::unique_ptr<globjects::Shader>              m_vertexShader;
```

Comment:
I presume your codes misses the macOS shader version replacements via shader templates entirely?
For an example, see https://github.com/cginternals/globjects/blob/primitive_memory_management/source/examples/programpipelines/main.cpp#L85.